### ISSUE TITLE
Source selection fails when there are no overlapping injection sources

### ISSUE THREAD

**Issue by stevenstetzler** [lsst/source_injection#41]
Source selection will fail at https://github.com/lsst/source_injection/blob/w.2024.33/python/lsst/source/injection/inject_base.py#L512 when there are no input injection sources. The composed injection catalog to be cleaned will have length 0 and when the selection occurs, one will run into a key error like so:
```
config.selection = "np.isin(injection_catalog['EXPNUM'], {visit}) # select on visit"
```

```
INFO 2024-08-20T11:54:54.415-07:00 lsst.inject_exposure (inject_exposure:{instrument: 'DECam', detector: 10, visit: 845324, band: 'VR', day_obs: 20190401, physical_filter: 'VR DECam c0007 6300.0 2600.0'})(inject_base.py:386) - Retrieved 0 injection sources from 0 HTM trixels.
ERROR 2024-08-20T11:54:54.419-07:00 lsst.ctrl.mpexec.singleQuantumExecutor (inject_exposure:{instrument: 'DECam', detector: 10, visit: 845324, band: 'VR', day_obs: 20190401, physical_filter: 'VR DECam c0007 6300.0 2600.0'})(singleQuantumExecutor.py:290) - Execution of task 'inject_exposure' on quantum {instrument: 'DECam', detector: 10, visit: 845324, band: 'VR', day_obs: 20190401, physical_filter: 'VR DECam c0007 6300.0 2600.0'} failed. Exception KeyError: 'EXPNUM'
ERROR 2024-08-20T11:54:54.424-07:00 lsst.ctrl.mpexec.mpGraphExecutor ()(mpGraphExecutor.py:533) - Task <inject_exposure dataId={instrument: 'DECam', detector: 10, visit: 845324, band: 'VR', day_obs: 20190401, physical_filter: 'VR DECam c0007 6300.0 2600.0'}> failed; processing will continue for remaining tasks.
Traceback (most recent call last):
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/ctrl_mpexec/g4fa1592e7e+7a7528f8bd/python/lsst/ctrl/mpexec/mpGraphExecutor.py", line 494, in _executeQuantaInProcess
    _, quantum_report = self.quantumExecutor.execute(task_node, qnode.quantum)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/ctrl_mpexec/g4fa1592e7e+7a7528f8bd/python/lsst/ctrl/mpexec/singleQuantumExecutor.py", line 176, in execute
    result = self._execute(task_node, quantum)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/ctrl_mpexec/g4fa1592e7e+7a7528f8bd/python/lsst/ctrl/mpexec/singleQuantumExecutor.py", line 288, in _execute
    self.runQuantum(task, quantum, task_node, limited_butler)
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/ctrl_mpexec/g4fa1592e7e+7a7528f8bd/python/lsst/ctrl/mpexec/singleQuantumExecutor.py", line 517, in runQuantum
    task.runQuantum(butlerQC, inputRefs, outputRefs)
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/source_injection/gceed50f6c5+0527dc3073/python/lsst/source/injection/inject_visit.py", line 129, in runQuantum
    outputs = self.run(**{key: value for (key, value) in inputs.items() if key in input_keys})
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/source_injection/gceed50f6c5+0527dc3073/python/lsst/source/injection/inject_base.py", line 231, in run
    injection_catalog = self._clean_sources(injection_catalog, input_exposure)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/share/eups/Linux64/source_injection/gceed50f6c5+0527dc3073/python/lsst/source/injection/inject_base.py", line 512, in _clean_sources
    selected = eval(self.config.selection.format(visit=visit))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 1, in <module>
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/lib/python3.11/site-packages/astropy/table/table.py", line 2061, in __getitem__
    return self.columns[item]
           ~~~~~~~~~~~~^^^^^^
  File "/nvme/users/stevengs/opt_lsst/conda/envs/lsst-scipipe-8.0.0/lib/python3.11/site-packages/astropy/table/table.py", line 264, in __getitem__
    return OrderedDict.__getitem__(self, item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'EXPNUM'
```

This is with version `w_2024_30`.

**Comment by stevenstetzler** [comment]
The expected behavior is for the task to succeed and write an output if
```
config.process_all_data_ids = True
```

**Comment by leeskelvin** [comment]
I've been able to confirm this `KeyError` at the USDF. This error is triggered when a unique set of circumstances are realized:

* The user is processing a dataset which does not have any overlapping injection sources;
* The user has requested to continue processing regardless of coverage overlap, by over-riding the config option `process_all_data_ids` to `True`;
* The user has requested that their input catalog be trimmed down according to some selection criteria by passing an argument into the config `selection`.

I've set up ticket [DM-45882](https://rubinobs.atlassian.net/browse/DM-45882) to capture effort to resolve this issue.
